addons/calendar_sms/models/calendar_alarm_manager.py

```diff
@@ -16,9 +16,8 @@ def _send_reminder(self):
         if not events_by_alarm:
             return
 
-        event_ids = list(set(event_id for event_ids in events_by_alarm.values() for event_id in event_ids))
-        events = self.env['calendar.event'].browse(event_ids)
-        alarms = self.env['calendar.alarm'].browse(events_by_alarm.keys())
-        for event in events:
-            alarm = event.alarm_ids.filtered(lambda alarm: alarm.id in alarms.ids)
-            event._do_sms_reminder(alarm)
+        all_events_ids = list(set(event_id for event_ids in events_by_alarm.values() for event_id in event_ids))
+        for alarm_id, event_ids in events_by_alarm.items():
+            alarm = self.env['calendar.alarm'].browse(alarm_id).with_prefetch(events_by_alarm.keys())
```

Comment:
```suggestion
            alarm = self.env['calendar.alarm'].browse(alarm_id).with_prefetch(list(events_by_alarm.keys()))
```
In practice this works because it's only ever used as an iterable of integers but `with_prefetch` normally expects a list and I'd air towards safety